chain/transaction.go

```diff
@@ -105,69 +102,106 @@ func (t *Transaction) Sign(
 	return UnmarshalTx(p, actionCodec, authCodec)
 }
 
-// Verify that the transaction was signed correctly.
-func (t *Transaction) Verify(ctx context.Context) error {
-	msg, err := t.UnsignedBytes()
-	if err != nil {
-		// Should never occur because populated during unmarshal
-		return err
+func (t *TransactionData) Expiry() int64 { return t.Base.Timestamp }
+
+func (t *TransactionData) MaxFee() uint64 { return t.Base.MaxFee }
+
+func (t *TransactionData) Marshal(p *codec.Packer) error {
+	if len(t.unsignedBytes) > 0 {
+		p.PackFixedBytes(t.unsignedBytes)
+		return p.Err()
+	}
+	return t.marshal(p)
+}
+
+func (t *TransactionData) marshal(p *codec.Packer) error {
+	t.Base.Marshal(p)
+	return t.Actions.marshalInto(p)
+}
+
+type Actions []Action
+
+func (a Actions) Size() (int, error) {
+	var size int
+	for _, action := range a {
+		actionSize, err := GetSize(action)
+		if err != nil {
+			return 0, err
+		}
+		size += consts.ByteLen + actionSize
+	}
+	return size, nil
+}
+
+func (a Actions) marshalInto(p *codec.Packer) error {
+	p.PackByte(uint8(len(a)))
+	for _, action := range a {
+		p.PackByte(action.GetTypeID())
+		err := marshalInto(action, p)
+		if err != nil {
+			return err
+		}
 	}
-	return t.Auth.Verify(ctx, msg)
+	return nil
 }
 
-func (t *Transaction) Bytes() []byte { return t.bytes }
+type Transaction struct {
+	TransactionData
+
+	Auth Auth `json:"auth"`
 
-func (t *Transaction) Size() int { return t.size }
+	bytes     []byte
+	size      int
+	id        ids.ID
+	stateKeys state.Keys
+}
 
-func (t *Transaction) ID() ids.ID { return t.id }
+func (s *Transaction) Bytes() []byte { return s.bytes }
```

Comment:
Can we change from using `(s *Transaction)` to `(t *Transaction)` to follow the convention of using the lowercase first letter of the type?